src/LiquidityPool.sol

```diff
@@ -462,10 +464,21 @@ contract LiquidityPool is Initializable, OwnableUpgradeable, UUPSUpgradeable, IL
         emit Rebase(getTotalPooledEther(), eETH.totalShares());
     }
 
+    function payProtocolFees(uint128 _protocolFees) public {
+        if (msg.sender != address(etherFiAdminContract)) revert IncorrectCaller();
+        uint256 treasuryShares = sharesForAmount(_protocolFees);
+        totalValueOutOfLp = uint128(int128(totalValueOutOfLp) + _protocolFees);
+        eETH.mintShares(treasury, treasuryShares);
+    }
```

Comment:
I think `totalValueOutOfLp` must be calculated first, and then `treasuryShares`

plz verify with tests

```
    function payProtocolFees(uint128 _protocolFees) public {
        if (msg.sender != address(etherFiAdminContract)) revert IncorrectCaller();
        totalValueOutOfLp = uint128(int128(totalValueOutOfLp) + _protocolFees);
        uint256 treasuryShares = sharesForAmount(_protocolFees);
        eETH.mintShares(treasury, treasuryShares);
    }
```

Reply:
oohhhh good catch